scanner/decisions/0005-scannerv4-read-cvss-data.md

```diff
@@ -9,18 +9,24 @@ Accepted
+
+The NVD JSON feed offers bundled data on a yearly basis, spanning from 2002 to the present year. A benefit is that only two requests are needed in the workflow for each year (one for the meta file and one for the data bundle). However, a limitation is the absence of filtering capabilities, such as by CVSS score or modification date. In contrast, the NVD API allows data retrieval through specific HTTP URL parameters, such as publish date, modified date, or CVSS score, enabling more precise data acquisition.
+
 
 ## Decision
 
-The current NVD CVSS GitHub Workflow will be updated to fetch data from the NVD CVE API instead of NVD Json feeds while maintaining the freshness of the NVD CVSS data bundle in Google Storage. It will categorize the downloaded data by CVSS V3 severity levels, and these categorized data will be archived into individual zip files, such as `severity-low.zip`. The format of the JSON is the same as before.
+The current NVD CVSS GitHub Workflow will be updated to fetch data from the NVD CVE API instead of NVD Json feeds while maintaining the freshness of the NVD CVSS data bundle in Google Storage. It will categorize the downloaded data by CVSS V3 severity levels, and these categorized data will be archived into individual zip files, such as `severity-low.zip`. The format of the JSON is the same as before.  
+
+We've chosen to base our API requests on CVSS V3 severity to ensure all data integrated into Claircore possesses valid CVSS V3 metrics. Consequently, retrieving data without CVSS V3 information becomes redundant.
 
 The CVSS Updater in Central will be modified to download zip files corresponding to four distinct severity levels and write them into a single zip file stored in Central's file system. This eliminates the need for JSON parsing in the updater. ScannerV4 will pull the zip bundle and populate the Matcher DB with the enrichment data.
 
 ## Consequences
 
-Transitioning from the NVD JSON feed to the NVD CVE API introduces certain constraints in data retrieval within the GitHub Action workflow, such as rate limiting and pagination. As a result, we will encounter a higher number of smaller, segmented JSON files.
-Changing from the NVD JSON feed to the NVD CVE API also give us the conveniences of no need to filter out data with out valid CVSS V3 metrics. 
+One drawback of the NVD CVE API is its rate limit: only 5 requests are allowed per 30 seconds without an API Key. This restriction necessitates that our workflow fetch data at a slower pace. Additionally, transitioning from the NVD JSON feed to the NVD CVE API brings in pagination limitations. For example, when fetching data with a "low" severity, the API provides a maximum of 2000 CVEs per response. This means multiple requests are required to retrieve the complete dataset for "low" severity. The result is an increased number of smaller JSON files. Consequently, we need to compress all JSON files for a particular severity into a single zip archive.
```

Comment:
> This restriction necessitates that our workflow fetch data at a slower pace

I [made one page
request](https://services.nvd.nist.gov/rest/json/cves/2.0?cvssV3Severity=CRITICAL&startIndex=0),
and the results say:

```
{"resultsPerPage":2000,"startIndex":0,"totalResults":21470...
```

That's roughly ten pages per severity. It would take ~1 minute to get all pages at ~0.15 pages per second (1 request / 6s). Fine.

It might be rate limiting is more nuanced than this, but I think it's fine to go for it and iterate.

Reply:
I think it may be worth asking if we can get an API key so we aren't as limited: https://nvd.nist.gov/developers/request-an-api-key

^ before filling that out we'd definitely need to ask for permission

Reply:
Or that, although I don't think it blocks the ADR at this point.